### PR TITLE
Use filesystem service for repositories

### PR DESCRIPTION
**Pull request by jbout**

### CODE REVIEW

File: common/oatbox/filesystem/FileSystemService.php [hunk 1]
```diff
@@ -53,7 +53,28 @@ public function getFileSystem($id)
...
+     * Create a new local file system
+     * 
+     * @param string $id
+     * @param string $path
+     * @return Filesystem
+     */
+    public function createLocalFileSystem($id)
+    {
+        $path = $this->getOption(self::OPTION_FILE_PATH).\helpers_File::sanitizeInjectively($id);
+        $success = $this->addLocalFileSystem($id, $path);
```
Comment (antoinerobin) [inline common/oatbox/filesystem/FileSystemService.php]: this method doesn't exist it should be ```registerLocalFileSystem```
Do you really need to get the return variable if you don't do anything with it ?

File: common/oatbox/filesystem/FileSystemService.php [hunk 2]
```diff
@@ -53,7 +53,28 @@ public function getFileSystem($id)
         return $this->filesystems[$id];
     }
     
-    public function addLocalFileSystem($id, $path)
+    /**
+     * Create a new local file system
+     * 
+     * @param string $id
+     * @param string $path
```
Comment (antoinerobin) [inline common/oatbox/filesystem/FileSystemService.php]: this is not a param

File: core/kernel/fileSystem/class.FileSystemFactory.php [hunk 3]
```diff
@@ -65,6 +67,11 @@ public static function createFileSystem( core_kernel_classes_Resource $type, $ur
         ));
         core_kernel_fileSystem_Cache::flushCache();
         
+        $serviceManager = ServiceManager::getServiceManager();
+        $fsm = $serviceManager->get(FileSystemService::SERVICE_ID);
+        $fsm->addLocalFileSystem($resource->getUri(), $path);
```
Comment (antoinerobin) [inline core/kernel/fileSystem/class.FileSystemFactory.php]: this method has been replaced by ```registerLocalFileSystem```

**Comment by antoinerobin** [comment]
```addLocalFileSystem``` is still used in ```tao_models_classes_FileSourceService::addLocalSource```

**Comment by jbout** [comment]
depends on https://github.com/oat-sa/tao-core/pull/696

**Pull request merged**